compiler/src/iree/compiler/Codegen/Common/GPU/GPUReuseSharedMemoryAllocs.cpp

```diff
@@ -45,6 +45,10 @@ static LivenessRange getLivenessRange(memref::AllocOp alloc,
         workList.push_back(user);
         continue;
       }
+      // Skip deallocations as they don't affect the liveness range.
+      if (isa<memref::DeallocOp>(user)) {
```

Comment:
Could we just use `alloca` for shared memory and not have deallocs at all?

Reply:
We could, but that's a decent amount of churn for not much benefit that I see. The deallocs are dropped at a later point though, I could just move that pattern up earlier.

Reply:
dropping a dealloc is basically saying the alloc is really an alloca. I understand the churn is an issue, but this is precisely the difference between alloca and alloc. We are really abusing alloc here.

Reply:
Yeah that is true... ok I can look into changing it.

Reply:
So most of the LLVMGPU side looks to work fine with `alloca`, but SPIR-V only supports `memref.alloc` with workgroup scope. I'll see if I can keep support for both for the time being then.

Reply:
This PR makes the switch for the two main pipelines we're developing: https://github.com/iree-org/iree/pull/20167

Some of the CUDA stuff is relying on functions that only support alloc upstream that I don't feel like going and changing.